app/subscribers/solidus_stripe/webhook/charge_subscriber.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "solidus_stripe/money_to_stripe_amount_converter"
+
+module SolidusStripe
+  module Webhook
+    # Handlers for Stripe charge events.
+    class ChargeSubscriber
+      include Omnes::Subscriber
+      include MoneyToStripeAmountConverter
+
+      handle :"stripe.charge.refunded", with: :refund_payment
+
+      # Refunds a payment.
+      #
+      # Creates a `Spree::Refund` for the payment associated with the
+      # webhook event.
+      #
+      # The event's `amount_refunded` field on Stripe contains the total amount
+      # refunded for the payment, including previous ones. We need to check that
+      # against the last total refunded amount on the payment to get the actual
+      # amount refunded by the current event.
+      #
+      # The `Spree::RefundReason` with `SolidusStripe::Config.refund_reason_name`
+      # as name is used as the created refund's reason.
+      #
+      # Notice that, at this point, we have no way to distinguish between
+      # multiple occurrences of the same event.
+      #
+      # @param event [SolidusStripe::Webhook::Event]
+      def refund_payment(event)
+        event.data.object.to_hash => {
+          amount_refunded: new_stripe_total,
+          payment_intent: payment_intent_id,
+          currency:
+        }
```

Comment:
In case anyone else is wondering what this 👽 syntax is, it is the same as:

```rb
new_stripe_total = event.data.object.to_hash[:amount_refunded]
payment_intent = event.data.object.to_hash[:payment_intent_id]
currency = event.data.object.to_hash[:currency]
```

or, maybe better:

```rb
new_stripe_total, payment_intent, currency = event.data.object.to_hash.values_at(
  :amount_refunded, :payment_intent_id, :currency
)
```